process_preds.Rmd

```diff
@@ -0,0 +1,116 @@
+---
+title: "Code to process predictors for urban niche model"
+author: "Mikko Jimenez, Josh Carrell, and Caitlin Mothes"
+date: "'r Sys.Date()'"
+output: html_notebook
+---
+
+## Process predictors
+
+As part of a broader pipeline for running a niche model in urban settings, this script is designed to provide a flexible process for reading in predictor variables, setting them to a consistent coordinate reference system, and cropping them to an extent of interest. 
+
+Note: it is currently unclear if predictors will be expected to be downloaded prior to running this script or if we will be pulling predictors directly from an API. I've written code for each situation below. 
+
+## set up environment
+```{r}
+source("setup.R")
+
+# read in spatial layer that includes full extent of interest, in this case a boundary .shp
+# having issues with symlink, run by Caitlin
+template <- read_sf("data/nature_in_the_city/gis/ft_collins_GMA_boundary.shp")
+
+# list of paths to predictors (shapefiles, .tifs, etc.)
+predictor_paths <- c("data/input_raw/Hydrology/Hydrology.shp",
+                     "data/input_raw/nlcd_tcc_CONUS_2021_v2021-4/nlcd_tcc_conus_2021_v2021-4.tif")
+
+# if needed, quick plot just to make sure your file looks right
+#plot(template)
+```
+
+## set the 'template' crs and extent and write a function that matches these for predictors
+```{r}
+# set the CRS and extent from the template shapefile
+template_crs <- st_crs(template)$proj4string
+template_extent <- st_bbox(template)
+
+# write function that reads in predictors and matches CRS and extent to the template
+# I assume we'll mostly be using .shps and .tifs, but my thought process is that we can add to this function if other file types come along 
+process_file <- function(file_path, template_crs, template_extent) {
+  if (grepl("\\.shp$", file_path, ignore.case = TRUE)) {
+    # Read in shapefile
```

Comment:
For shapefiles, we will want to rasterize them since model inputs need to be rasters. That could be the last step here after cropping, and then the resolution within the `terra::rasterize()` function will be the user-given value.